reference/events.rst

```diff
@@ -0,0 +1,244 @@
+method::
+
+    use Symfony\Component\HttpKernel\Event\GetResponseForControllerResultEvent;
+    use Symfony\Component\HttpFoundation\Response;
+
+    public function onKernelView(GetResponseForControllerResultEvent $event)
+    {
+        $val = $event->getControllerResult();
+        $response = new Response();
+
+        // ... somehow customize the Response from the return value
+
+        $event->setResponse($response);
+    }
+
+.. seealso::
+
+    Read more on the :ref:`kernel.view event <component-http-kernel-kernel-view>`.
+
+``kernel.response``
+~~~~~~~~~~~~~~~~~~~
+
+**Event Class**: :class:`Symfony\\Component\\HttpKernel\\Event\\FilterResponseEvent`
+
+The purpose of this event is to allow other systems to modify or replace the
+``Response`` object after its creation::
+
+    public function onKernelResponse(FilterResponseEvent $event)
+    {
+        $response = $event->getResponse();
+
+        // ... modify the response object
+    }
+
+The FrameworkBundle registers several listeners:
+
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\ProfilerListener`
+    Collects data for the current request.
+
+:class:`Symfony\\Bundle\\WebProfilerBundle\\EventListener\\WebDebugToolbarListener`
+    Injects the Web Debug Toolbar.
+
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\ResponseListener`
+    Fixes the Response ``Content-Type`` based on the request format.
+
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\EsiListener`
+    Adds a ``Surrogate-Control`` HTTP header when the Response needs to be parsed
+    for ESI tags.
+
+.. seealso::
+
+    Read more on the :ref:`kernel.response event <component-http-kernel-kernel-response>`.
+
+These are the built-in Symfony listeners related to this event:
+
+===================================================================================  ========
+Listener Class Name                                                                  Priority
+===================================================================================  ========
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\EsiListener`                  0
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\ResponseListener`             0
+:class:`Symfony\\Bundle\\SecurityBundle\\EventListener\\ResponseListener`            0
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\ProfilerListener`             -100
+:class:`Symfony\\Bundle\\FrameworkBundle\\EventListener\\TestSessionListener`        -128
+:class:`Symfony\\Bundle\\WebProfilerBundle\\EventListener\\WebDebugToolbarListener`  -128
+:class:`Symfony\\Component\\HttpKernel\\EventListener\\StreamedResponseListener`     -1024
+===================================================================================  ========
+
+``kernel.terminate``
+~~~~~~~~~~~~~~~~~~~~
+
+**Event Class**: :class:`Symfony\\Component\\HttpKernel\\Event\\PostResponseEvent`
+
+The purpose of this event is to perform "heavier" tasks after the response
```

Comment:
this `heavier` is rather subjective. Maybe explaining tasks that can wait so that we can return a response?